lib/device.js

```diff
@@ -251,6 +256,169 @@ const util = require('util'),
+                _captureFile,
+                _copyToLocal
+            ],  function(err, results) {
+                log.verbose("device#capture()", "err: ", err, "results:", results);
+                const resultTxt = "Create " + chalk.green(options.captureFileName) + " to " + options.destinationPath +"\nSuccess";
+
+                // clean up /tmp/capture directory in target device
+                if (options.targetCaptureDir) {
+                    _removeCaptureDir(function finish (removeErr){
+                        next(err ? err : removeErr, resultTxt);
+                    });
+                } else {
+                    next(err, resultTxt);
+                }
+            });
+
+            function _makeSession(next) {
+                options.nReplies = 1; // -n 1
+                options.session = new novacom.Session(options.device, next);
+            }
+
+            function _makeCaptureOption(next) {
+                log.info("device#capture#_makeCaptureOption()");
+                let captureFileName = "",
+                    destinationPath = "",
+                    captureFormat = "PNG"; // PNG is default format
+
+                if (options.outputPath === null){
+                    captureFileName = _createDateFileName();
+                    destinationPath = path.resolve('.');
+                } else {
+                    // Get directory path, file path, file Extenstion from given path
+                    const parseDirPath = path.dirname(options.outputPath),
+                        parseBase = path.parse(options.outputPath).base;
+
+                    let parseExt = path.parse(options.outputPath).ext;
+                    parseExt = parseExt.split('.').pop();
+
+                    log.info("device#capture#_makeCaptureOption():" + " dir name : " + parseDirPath 
+                        + " ,file name : " + parseBase +" ,inputFormat : " +  parseExt);
+
+                    // if given path has extension
+                    if (parseBase && parseExt) {
+                        if (parseExt === "png" || parseExt === "bmp" || parseExt === "jpg") {
+                            captureFormat = parseExt.toUpperCase();
+                        } else {
+                            return setImmediate(next, errHndl.changeErrMsg("INVALID_CAPTURE_FORMAT"));
+                        }
+
+                        captureFileName = parseBase;
+                        destinationPath = path.resolve(parseDirPath);
+                    } else if (parseBase && (parseExt === "")) {
+                        // it's directory path. Use it as destination dir Path
+                        captureFileName = _createDateFileName();
+                        destinationPath = path.resolve(options.outputPath);
+                    }
+                }
+
+                options.captureFormat = captureFormat;
+                options.captureFileName = captureFileName;
+                options.captureDirPath = "/tmp/capture",
```

Comment:
How about writing the folder name in more detail in case it won't be deleted?

Reply:
I change it to /tmp/aresCapture